underfs/hdfs/src/main/java/alluxio/underfs/hdfs/HdfsUnderFileInputStream.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.hdfs;
+
+import alluxio.underfs.SeekableUnderFileInputStream;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+
+import java.io.IOException;
+
+/**
+ * The input stream of HDFS as under filesystem. This input stream supports seeking and can be
+ * cached for reuse.
+ */
+public class HdfsUnderFileInputStream extends SeekableUnderFileInputStream {
+
+  HdfsUnderFileInputStream(FSDataInputStream in) {
+    super(in);
+  }
+
+  @Override
+  public void seek(long position) throws IOException {
+    ((FSDataInputStream) in).seek(position);
+  }
+
+  @Override
+  public long getPos() {
+    try {
+      return ((FSDataInputStream) in).getPos();
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to get position " + e);
```

Comment:
Is this a good idea, this will kill the thread arbitrarily?

Reply:
hmm, the other option is to return -1. If the underfs throws an exception then it should be propagated to the caller since the stat is incorrupted?

Reply:
For client calls I think its fine. I am concerned that this will be called from non-client threads (ie. in worker background threads) and then cause services to die. Though we should address that type of problem by making the those threads more robust.

Reply:
In that case, I would change the interface to throw IOException. What do you think?

Reply:
I don't have a strong preference for either solution, both have pros/cons, up to you.